coreax/metrics.py

```diff
@@ -428,3 +429,200 @@ def sum_weighted_pairwise_distances(
+        kernel function :math:`k: \mathbb{R}^p \times \mathbb{R}^p \rightarrow
+        \mathbb{R}` on the response space
+    :param num_feature_dimensions: An integer representing the dimensionality of the
+        features :math:`x`
+    :param regularisation_parameters: A  :math:`1 \times 2` array of regularisation
+        parameters corresponding to the original dataset :math:`\mathcal{D}^{(1)}` and
+        the coreset :math:`\mathcal{D}^{(2)}` respectively
+    :param precision_threshold: Positive threshold we compare against for precision
+    """
+
+    def __init__(
+        self,
+        feature_kernel: coreax.kernel.Kernel,
+        response_kernel: coreax.kernel.Kernel,
+        num_feature_dimensions: int,
+        regularisation_parameters: ArrayLike,
+        precision_threshold: float = 1e-6,
+    ):
+        """Calculate conditional maximum mean discrepancy between two datasets."""
+        self.feature_kernel = feature_kernel
+        self.response_kernel = response_kernel
+        self.num_feature_dimensions = num_feature_dimensions
+        self.regularisation_parameters = regularisation_parameters
+        self.precision_threshold = precision_threshold
+
+        # Initialise parent
+        super().__init__()
+
+    # Disable pylint warning for arguments-renamed as the use of x and y as arguments
+    # for different datasets is bound to create misunderstandings when we are
+    # considering supervised data.
+    # pylint: disable=arguments-renamed
+    def compute(
+        self,
+        dataset_1: ArrayLike,
+        dataset_2: ArrayLike,
+        block_size: None = None,
+        weights_x: None = None,
+        weights_y: None = None,
+    ) -> Array:
+        r"""
+        Calculate conditional maximum mean discrepancy.
+
+        :param dataset_1: The original dataset :math:`\mathcal{D}^{(1)} =
+            \{(x_i, y_i)\}_{i=1}^n` of ``n`` pairs with :math:`x\in\mathbb{R}^d` and
+            :math:`y\in\mathbb{R}^p`, responses should be concatenated after the
+            features
+        :param dataset_2: Dataset :math:`\mathcal{D}^{(2)} = \{(\tilde{x}_i,
+            \tilde{y}_i)\}_{i=1}^n` of ``m`` pairs with :math:`\tilde{x}\in\mathbb{R}^d`
+            and :math:`\tilde{y}\in\mathbb{R}^p`, responses should be
+            concatenated after the features
+        :param block_size: :data:`None`, included for compatibility reasons
+        :param weights_x: :data:`None`, included for compatibility reasons
+        :param weights_y: :data:`None`, included for compatibility reasons
+        :return: Conditional maximum mean discrepancy as a 0-dimensional array
+        """
+        # Make sure that compatibility params are None
+        assert block_size is None, "CMMD computation does not support blocking"
+        assert weights_x is None, "CMMD computation does not support weights"
+        assert weights_y is None, "CMMD computation does not support weights"
```

Comment:
A refactor of the `Metric` abstract might be something for future discussion, so I would encourage raising a feature request ticket if you think it would be useful. This might be covered if we introduced weighted datastructures/arrays.

Reply:
Yeah I think this is something that will make sense to change in the future, I'll set up an issue for it. Should I remove the assert statements for now?